TP2_sofíagritti.Rmd

```diff
@@ -0,0 +1,174 @@
+
+### 2) Existe una diferencia de encuestadora, que podría sesgar los resultados estadísticos.
+
+```{r}
+resumen <- encuestas %>%
+  group_by(Encuestadora, month(Fecha)) %>%
+  summarize(
+    Promedio_Todos = mean(Todos),
+    Desviacion_Todos = sd(Todos),
+    Promedio_Juntos = mean(Juntos),
+    Desviacion_Juntos = sd(Juntos),
+    Promedio_Libertad_Avanza = mean(Libertad_Avanza),
+    Desviacion_Libertad_Avanza = sd(Libertad_Avanza),
+    Promedio_Consenso_Federal = mean(Consenso_Federal),
+    Desviacion_Consenso_Federal = sd(Consenso_Federal),
+    Promedio_FIT = mean(FIT),
+    Desviacion_FIT = sd(FIT)
+  )
+
+resumen <- resumen[order(resumen$`month(Fecha)`), ]
+tabla_formateada <- kable(resumen, format = "html", table.attr = "class='styled-table'")
+tabla_formateada
+```
+
+En respuesta a la pregunta, podemos ver la gran diferencia en el promedio que obtienen los partidos mes a mes según la encuestadora, permitiendonos plantear la idea de que puede existir un sesgo en base al formato de la medición.
+
+### 3) Existe una paridad entre los 3 partidos mayoritarios que se mantiene en el tiempo por lo que las encuestas no permiten predecir el ballotaje.
+
+```{r}
+encuestas$Fecha <- as.Date(encuestas$Fecha, format = "%d/%m/%Y")
+encuestas$Mes <- format(encuestas$Fecha, "%Y-%m")
+promedio_mes <- encuestas %>%
+  group_by(Mes) %>%
+  summarise(
+    Promedio_Todos = mean(Todos),
+    Promedio_Juntos = mean(Juntos),
+    Promedio_Libertad_Avanza = mean(Libertad_Avanza)
+  )
+print(promedio_mes)
+```
+Como se puede observar, según las encuestas relevadas, se espera una elección de 3 tercios, por lo que es imposible determinar cuáles van a ser participes del ballotaje según este resultado. Claramente, si se mantienen estos datos, si se puede garantizar la probabilidad de un ballotaje porque ninguno cumple los requisitos de mayoría absoluta relativa para obtener resultados en primera vuelta.
+
+### 4) Cómo fluctúan los votos en los partidos.
+```{r}
+encuestas$Fecha <- as.Date(encuestas$Fecha, format = "%d/%m/%Y")
+encuestas$Mes <- format(encuestas$Fecha, "%Y-%m")
+encuestas_sin_nulos <- encuestas[complete.cases(encuestas$Todos), ]
+intervalos_confianza <- encuestas_sin_nulos %>%
+  group_by(Mes) %>%
+  summarise(
+    Promedio_Todos = mean(Todos),
+    Promedio_Juntos = mean(Juntos),
+    Promedio_Libertad_Avanza = mean(Libertad_Avanza),
+    IC_Todos = ifelse(n() > 1, paste0("[", formatC(t.test(Todos)$conf.int[1], format = "f", digits = 2), " - ", formatC(t.test(Todos)$conf.int[2], format = "f", digits = 2), "]"), "N/A"),
+    IC_Juntos = ifelse(n() > 1, paste0("[", formatC(t.test(Juntos)$conf.int[1], format = "f", digits = 2), " - ", formatC(t.test(Juntos)$conf.int[2], format = "f", digits = 2), "]"), "N/A"),
+    IC_Libertad_Avanza = ifelse(n() > 1, paste0("[", formatC(t.test(Libertad_Avanza)$conf.int[1], format = "f", digits = 2), " - ", formatC(t.test(Libertad_Avanza)$conf.int[2], format = "f", digits = 2), "]"), "N/A")
+  )

```

Comment:
Esto lo podés hacer mucho más corto usando la tabla larga. Además, podés usar el paquete broom para que te de los valores del t.test en la tabla: 

```r
encuestas_alargadas |> 
  group_by(mes = format(Fecha, "%Y-%m"), Partido) |> 
  filter(n() > 1) |> 
  summarise(broom::tidy(t.test(Intencion_voto)))
```

(Notá que el filter se hace sobre la tabla agrupada por lo que el `n()` se calcula para cada grupo). 

Esto te da estimaciones de la media y sus intervalos de confianza inferior y superior para cada partido en cada mes (cuando hay suficientes encuestas). 

Con eso